### PR TITLE
DM-11390: Plug prototype pipeline script into verify_ap framework

### PR DESCRIPTION
**Pull request by mrawls**

### CODE REVIEW

File: python/lsst/ap/verify/pipeline_driver.py [hunk 1]
```diff
@@ -47,14 +54,15 @@ def __init__(self):
...
+        self.add_argument("-j", "--processes", default=1, type=int, 
+                          help="Number of processes to use. Not yet implemented.")
 
 
 class MeasurementStorageError(RuntimeError):
     pass
 
 
-def _update_metrics(self, metadata, job):
+def _update_metrics(metadata, job):
```
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: This was actually a bug I found; it looked to me like you had removed `self` from all the other used-to-be classes in master but this one was accidentally left behind. No harm done.

[the remark this answers is not in the document]
Comment (kfindeisen) [reply python/lsst/ap/verify/pipeline_driver.py]: D'oh!

File: python/lsst/ap/verify/pipeline_driver.py [hunk 2]
```diff
@@ -30,8 +30,15 @@
...
 import lsst.log
+import lsst.ap.pipe as ap_pipe
 from lsst.verify import Job
 
+# Names of directories to be created in specified output location
+INGESTED_DIR = 'ingested'
+CALIBINGESTED_DIR = 'calibingested'
+PROCESSED_DIR = 'processed'
+DIFFIM_DIR = 'diffim'
+OUTPUT_DIRS = [INGESTED_DIR, CALIBINGESTED_DIR, PROCESSED_DIR, DIFFIM_DIR]
```
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: I couldn't figure out a smart way for these global variable type things to be known by `ap_verify` when I'm just using a single function from `ap_pipe`. This at least seemed better than defining them in _all_ the `ap_pipe` functions redundantly, but I'm open to suggestions.

File: python/lsst/ap/verify/pipeline_driver.py [hunk 3]
```diff
@@ -150,41 +162,11 @@ def _ingest_calibs(dataset, working_repo, metrics_job):
...
-    `pipeline.MeasurementStorageError`
-        Measurements were made, but `metrics_job` could not be updated
-        with them.
-    """
-    raise NotImplementedError
-
+    repo = ap_pipe.get_output_repos(working_repo, OUTPUT_DIRS)[0]
+    calib_repo = ap_pipe.get_output_repos(working_repo, OUTPUT_DIRS)[1]
+    calibdatafiles = ap_pipe.get_calibdatafiles(dataset.data_location, dataset.calib_location)
+    defectfiles = ap_pipe.get_defectfiles(dataset.data_location, dataset.defect_location, 'defects_2014-12-05.tar.gz')
```
[the remark this answers is not in the document]
Comment (kfindeisen) [reply python/lsst/ap/verify/pipeline_driver.py]: `data_location` = `<directory>/raw` and `calib_location` = `<directory>/calib`...

[the remark this answers is not in the document]
Comment (kfindeisen) [reply python/lsst/ap/verify/pipeline_driver.py]: Ok, so I was completely wrong. Hmm...

[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: We fixed this in person, yay!

File: python/lsst/ap/verify/pipeline_driver.py [hunk 4]
```diff
@@ -150,41 +162,11 @@ def _ingest_calibs(dataset, working_repo, metrics_job):
...
-    Raises
-    ------
-    `pipeline.MeasurementStorageError`
-        Measurements were made, but `metrics_job` could not be updated
-        with them.
-    """
-    raise NotImplementedError
-
+    repo = ap_pipe.get_output_repos(working_repo, OUTPUT_DIRS)[0]
+    calib_repo = ap_pipe.get_output_repos(working_repo, OUTPUT_DIRS)[1]
```
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: Fixed now, yay!

File: python/lsst/ap/verify/pipeline_driver.py [hunk 5]
```diff
@@ -119,8 +130,9 @@ def _ingest_raws(dataset, working_repo, metrics_job):
         Measurements were made, but `metrics_job` could not be updated
         with them.
     """
-    raise NotImplementedError
-
+    repo = ap_pipe.get_output_repos(working_repo, OUTPUT_DIRS)[0]
+    datafiles = ap_pipe.get_datafiles(dataset.data_location, dataset.data_location)
```
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: This has now been reworked in person, yay!

File: python/lsst/ap/verify/pipeline_driver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ap/verify/pipeline_driver.py]: We discussed this in person; we can't get rid of these entirely but they're handled slightly better now.

File: python/lsst/ap/verify/pipeline_driver.py [hunk 6]
```diff
@@ -245,8 +228,10 @@ def _difference(working_repo, dataId, parallelization, metrics_job):
         Measurements were made, but `metrics_job` could not be updated
         with them.
     """
-    raise NotImplementedError
-
+    processed_repo = ap_pipe.get_output_repos(working_repo, PROCESSED_DIR)
+    diffim_repo = ap_pipe.get_output_repos(working_repo, DIFFIM_DIR)
+    template = '410929'  # one g-band Blind15A40 visit for testing
```
Comment (kfindeisen) [inline python/lsst/ap/verify/pipeline_driver.py]: Maybe mention DM-11422 here?

**Pull request merged**